src/api/ledger/parse/orderbook-order.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const _ = require('lodash');
+const utils = require('./utils');
+const flags = utils.core.Remote.flags.offer;
+const parseAmount = require('./amount');
+
+function parseOrderbookOrder(order: Object): Object {
+  const direction = (order.Flags & flags.Sell) === 0 ? 'buy' : 'sell';
+  const takerGetsAmount = parseAmount(order.TakerGets);
+  const takerPaysAmount = parseAmount(order.TakerPays);
+  const quantity = (direction === 'buy') ? takerPaysAmount : takerGetsAmount;
+  const totalPrice = (direction === 'buy') ? takerGetsAmount : takerPaysAmount;
+
+  const specification = utils.removeUndefined({
+    direction: direction,
+    quantity: quantity,
+    totalPrice: totalPrice,
+    passive: ((order.Flags & flags.Passive) !== 0) || undefined
+  });
+  // "quality" is omitted intentionally as it corresponds to
```

Comment:
> You've changed the usual semantics

"Usual" according to who? This semantics may not be the way rippled does it, but it is the way the rest of the world does it.

> Is it actually a useful change

It's debatable. The way the FX market deals with this is to have a conventional ordering of each pair based on which currency is considered "stronger" so that you don't have two different books for basically the same thing. We probably want to go with that convention unless we are really convinced we have an objectively better option.

> You have it the wrong way around. 

I think I meant `/` as "slash" not division; I wasn't paying attention to the ordering.

> With partial executions, the ratio may slip.

This is a good point, maybe we will have to show the price. Maybe we can drop the totalPrice and show either "price" xor "inversePrice" depending on the case.